pkg/networking/middleware/useragent.go

```diff
@@ -0,0 +1,49 @@
+package middleware
+
+import (
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/snyk/go-application-framework/pkg/configuration"
+)
+
+type UserAgentMiddleware struct {
+	next               http.RoundTripper
+	config             configuration.Configuration
+	snykAppEnvironment *UserAgentInfo
+}
+
+func NewUserAgentMiddleware(
+	config configuration.Configuration,
+	roundTripper http.RoundTripper,
+	snykAppEnvironment *UserAgentInfo,
+) *UserAgentMiddleware {
+	return &UserAgentMiddleware{
+		next:               roundTripper,
+		config:             config,
+		snykAppEnvironment: snykAppEnvironment,
+	}
+}
+
+func (n *UserAgentMiddleware) RoundTrip(request *http.Request) (*http.Response, error) {
+	// Only add headers if the request is going to a Snyk API URL.
+	apiUrl := n.config.GetString(configuration.API_URL)
+	parsedUrl, err := url.Parse(apiUrl)
+	if err != nil {
+		return n.next.RoundTrip(request)
+	}
+
+	isSnykApi, err := IsSnykApi(apiUrl, request.URL, nil)
+	if err != nil || n.snykAppEnvironment == nil || !isSnykUrl(parsedUrl.Hostname()) || !isSnykApi {
```

Comment:
Doesn't `isSnykAPI` (or the previous name) encompass `isSnykUrl`?

Reply:
IsSnykUrl is whether it's "snyk.io/snykgov.io", IsSnykApi can be true for any URL given the right parameters.

But the naming is confusing so I'll change it

Reply:
How can something be a Snyk API without being part of `snyk.io` or `snykgov.io`? I think that would not be an allowed pattern.